src/eventProcessor/GoogleAnalyticsEventProcessor.js

```diff
@@ -50,12 +64,57 @@ class GoogleAnalyticsEventProcessor {
   constructor({
       'api_secret': apiSecret,
       'measurement_id': measurementId,
-      'measurement_url': measurementUrl = DEFAULT_MEASUREMENT_URL,
-      'client_id_expires': clientIdExpires = 2 * 365 * 24 * 60 * 60,
-      'automatic_params': userAutomaticParams = [],
+      'measurement_url': measurementUrl,
+      'client_id_expires': clientIdExpires,
+      'automatic_params': userAutomaticParams,
     } = {}) {
-    // TODO(kjgalvan):: log error if measurementUrl is default value and
-    // either apiSecret or measurementId are undefined.
+    if (apiSecret !== undefined && typeof apiSecret !== 'string') {
+      logging.log(
+        'Provided API Secret is not a valid string.',
+        logging.LogLevel.ERROR
+      );
+      apiSecret = undefined;
+    }
+    if (measurementId !== undefined && typeof measurementId !== 'string') {
+      logging.log(
+        'Provided Measurement ID is not a valid string.',
+        logging.LogLevel.ERROR
+      );
+      measurementId = undefined;
+    }
+    if (typeof measurementUrl !== 'string') {
+      if (measurementUrl !== undefined) {
+        logging.log(
+          'Provided Measurement URL is not a valid string. ' +
+              'Using default URL instead.',
+          logging.LogLevel.ERROR
+        );
+      }
+      measurementUrl = getDefaultMeasurementUrl();
+    }
+    if (clientIdExpires === undefined || isNaN(Number(clientIdExpires))) {
```

Comment:
isNaN(Number(clientIdExpires) should be sufficient

Reply:
I also am using these blocks to also set default values - I need to detect is client_id_expires was undefined as Number(undefined) actually becomes a number (0).